### PR TITLE
Details page

### LINKED ISSUE #2: 2 - create a article details page
## Value Proposition

**As a** user  
**I want to** see all details of a article  
**so that** I can better decide for a purchas
​

## Description

userStory_02-1
​
userStory_02-2

## Acceptance Criteria

- [ ] the article card should have a "I" icon
- [ ] all details should be displayed by clicking on the "I" icon
- [ ] the article page with all details should contain the following details:
      image, Artikel, Preis, Größe, Geschlecht, Zustand, Tierhaushalt, Raucherhaushalt, kurze Beschreibung, Entfernung
- [ ] the article details page should hav it`s own headline "Artikel"
- [ ] the article details page should have a icon "X" for closing the details and go back to homepage
      ​
## Tasks

- [x] create a feature-branch
- [x] create a `articleDetails` page
- [x] create a icon with a link to the `articleDetails`
- [x] the articleDetails should display in a new page (Modal)
- [x] link the icon to the articleDetails
- [x] render all details at articleDetails
- [x] create an "X" icon to close the page and go back to homepage
      ​
## Complexity

L

### PR DESCRIPTION
**Pull request by EugenMoser**

### CODE REVIEW

File: components/Header/index.js [hunk 1]
```diff
@@ -1,5 +1,18 @@
+import { useRouter } from "next/router";
+
 function Header() {
-  return <h1>Bazam</h1>;
+  const { pathname } = useRouter();
+
+  console.log(pathname);
```
Comment (mpagels) [inline components/Header/index.js]: Lonely console.log here ;-)

File: components/Header/index.js [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+  return (
+    <>
+      {pathname.startsWith("/articleDetails") ? (
+        <h1>Details</h1>
+      ) : (
+        <h1>Bazam</h1>
+      )}
```
Comment (mpagels) [inline components/Header/index.js]: Just a little improvement suggestion here: I would recommend putting the first check logic over the `return` statement. Something like this:

```js
const isArticleDetails = {pathname.startsWith("/articleDetails")
return <h1>{isArticleDetails ? "Details" : "Bazam"}</h1>
````

File: helpers/db.js [hunk 3]
```diff
@@ -4,4 +4,10 @@ function getAllArticles() {
   return articlesDb;
 }
 
-export default getAllArticles;
+function getArticlesById(id) {
+  console.log(id);
```
Comment (mpagels) [inline helpers/db.js]: Another console.log here

File: pages/index.js [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
-import styled from "styled-components";
 import useSWR from "swr";
 import { fetcher } from "../helpers/api";
 
 function Home() {
   const { data: articles, error } = useSWR("/api/articles", fetcher);
 
   if (!articles) {
-    return <h4>loading...</h4>;
+    return <h4>Artikel Details werden geladen...</h4>;
```
Comment (mpagels) [inline pages/index.js:9]: Why do you use a `h4` tag here? Why not start with `h1` ?

File: pages/articleDetails/[id]/index.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import Svg from "../../../components/Svg";
+import Link from "next/link";
+
+function articleDetails() {
+  const { query } = useRouter();
+  const { id } = query;
+  const { data: article, error } = useSWR(`/api/articles/${id}`, fetcher);
+
+  if (!article) return <p>Artikel Details werden geladen...</p>;
+  if (error) return <p>Fehler! keine Artikel Details gefunden.</p>;
```
Comment (mpagels) [inline pages/articleDetails/[id]/index.js]: Just thinking... Would it be better switching these two lines? Right now: If an error occurs, the article is not there, just line 15 is rendered and line 16 never gets rendered. But I might be wrong. But, thinking switching these two lines would be better if an error orrcurs.

```suggestion
  if (error) return <p>Fehler! keine Artikel Details gefunden.</p>;
  if (!article) return <p>Artikel Details werden geladen...</p>;
```

File: pages/articleDetails/[id]/index.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    size,
+    gender,
+    status,
+    animal,
+    smoker,
+    description,
+    distance,
+    image,
+  } = article;
+  console.log(animal, smoker);
```
Comment (mpagels) [inline pages/articleDetails/[id]/index.js]: console.log

File: pages/articleDetails/[id]/index.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      <StyledArticle>
+        <h3>Details zum Artikel: {name}</h3>
+        <Image
+          src={image}
+          alt={`Bild von ${name}`}
+          width={150}
+          height={200}
+        />
+        <ul>
+          <StyledList>Preis: {price} Euro</StyledList>
```
Comment (mpagels) [inline pages/articleDetails/[id]/index.js]: Seems strange reading Styled LIST, but having a listitem and not a list ;-) Could be confusing for future Eugen ;-) Recommend changing the name.

**Review by mpagels — APPROVED** [review]
Found some `console.log`s in here. 
And an empty file? [components/ButtonIcon/index.js](https://github.com/EugenMoser/capstone-project/pull/6/files#diff-15a6fd24f8ddeca8499f792298fbccd247d3125bc4190de7add8337b8528729b)
Do you need that in your production code?

Left some improvement ideas. But overall very good work here. Keep up!

File: pages/index.js (hunk 4 above)
Comment (EugenMoser) [reply pages/index.js:9]: h1 is in the headline ("Bazam", "Details"). I thought there must be only one `h1` tag for eache page

File: pages/articleDetails/[id]/index.js (hunk 7 above)
Comment (EugenMoser) [reply pages/articleDetails/[id]/index.js]: thanks for your great feedback Martin

**Pull request merged**